### DISCUSSION TITLE
Websocket help

### DISCUSSION THREAD

**Discussion by halioris** [WarmUpTill/SceneSwitcher#1065] [Q&A]
Looking for what I hope is easy help with websocket. I just downloaded and installed Advanced Scene Switcher with the intent of using it simply to receive a websocket message and run one of my Lua scripts using data from the message as parameters (so I don't have to figure out how to write my own websocket plugin).  I already have the websocket communication working from Powershell to OBS where I am sending in a message to change a scene name followed by another to trigger a hotkey by name and that works fine.  So my first test with Advanced Scene Switcher is just to show that it received a message.  So I set a trigger with a condition of Websocket and selected "Scene Switcher Request" in the dropdown (the default, and was surprised it didn't say Advanced Scene Switcher Request - hope thats not an issue).  For the action I just selected Switch to a standard scene.  When I send the message in nothing happens.  Some questions.
1) Do I need to do anything else in the condition?  
2) I have no idea what the "--enter text--" is in the condition.  Didn't see anything about it on the Wiki page for Websocket.
3) Once I actually get it to receive the messgae, what will be the best way to call a Lua script with data from the message.  What I was doing from Powershell was renaming a scene with the data in the scene name and then calling a haotkey that ran my script that parsed the scene name.  I don't like the scene thing and just want to get the data from the emssage and run my script.

For the record here is the message I am sending:
```
{
  "d": {
    "requestData": {
      "requestType": "AdvancedSceneSwitcherMessage",
      "vendorName": "AdvancedSceneSwitcher",
      "requestData": {
        "message": "testing"
      }
    },
    "requestType": "CallVendorRequest",
    "requestId": "62172596-bfa9-470a-827c-d66c2d6e74e9"
  },
  "op": 6
}
```
and here is the response I am getting:
```
{
   "d":{
      "requestId":"62172596-bfa9-470a-827c-d66c2d6e74e9",
      "requestStatus":{
         "code":100,
         "result":true
      },
      "requestType":"CallVendorRequest",
      "responseData":{
         "requestType":"AdvancedSceneSwitcherMessage",
         "responseData":{},
         "vendorName":"AdvancedSceneSwitcher"
      }
   },
   "op":7
}
```

Thanks for any help/thoughts, but really I'll just be happy if I can see the darn thing react to a message.

**Comment by halioris** [comment]
Ok on a hunch I put "testing" in my condition and my action fired.  Great, making progress.  So now how can I do something a little more complicated in the condition where I extract variable names and do something wiht them.  Hoping someone beats me to an answer before I spend hours trying and then looking through the source code to figure out what to do.

**Comment by WarmUpTill** [comment]
>I have no idea what the "--enter text--" is in the condition. Didn't see anything about it on the Wiki page for Websocket.

The condition will only evaluate to `true` if the received message matches the given text.
If you want it to match on any received data you will have to enable pattern matching and use the pattern `.*`.

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/c0ca60c5-f073-40f3-b232-c04f12694129)

> So now how can I do something a little more complicated in the condition where I extract variable names

You can use the "Variable" action to assign the received message to a variable.

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/12f0bca6-cd16-46d0-8446-04f7cf72a235)

Depending on how many arguments you have and how the message received is structured you can then use other options of the "Variable" action type to process them further.
For example, if you would be sending a json message you might do something similar to this to extract the fields and assign them to separate variables:

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/e306f60e-af43-45d8-af33-3a77ce4a3f82)

>do something wiht them

You can then reference those variables using the `${variable name here}` syntax in the "Run" action to run your lua script with the required arguments:

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/79bcfef2-6b9b-4c50-a5bf-85ac6cd5335f)

Hope that helps!
Let me know if you have any questions! :)
